orc8r/cloud/go/services/state/servicers/servicer.go

```diff
@@ -58,53 +58,52 @@ func (srv *stateServicer) GetStates(ctx context.Context, req *protos.GetStatesRe
 	return srv.searchStates(ctx, req)
 }
 
+// ReportStates from a gateway.
+// Always reports UnreportedStates as empty.
 func (srv *stateServicer) ReportStates(ctx context.Context, req *protos.ReportStatesRequest) (*protos.ReportStatesResponse, error) {
-	res := &protos.ReportStatesResponse{}
-	validatedStates, invalidStates, err := PartitionStatesBySerializability(req)
-	if err != nil {
-		return res, status.Error(codes.InvalidArgument, err.Error())
+	if err := validateReportStatesRequest(req); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	res.UnreportedStates = invalidStates
```

Comment:
Concerned about the possibility of saving a bunch of malformed crap in the backend without this logic. Have you shored up the read side of this service's client API to prevent bad code on the reporting path from tanking the read path?

I guess there's no getting around this without introducing a new RPC interface for validation. But that's out of our immediate scope here so maybe document the shortcoming here and we can bootcamp this or something.

Reply:
Yes handled at the readers, at some point we should add the capability to validate at the edge but for now I agree this is good enough